tools/accuracy_checker/accuracy_checker/metrics/coco_orig_metrics.py

```diff
@@ -109,84 +109,43 @@ def parameters(cls):
     def configure(self):
         self.threshold = get_or_parse_value(self.get_value_from_config('threshold'), COCO_THRESHOLDS)
 
-    @staticmethod
-    def generate_map_pred_label_id_to_coco_cat_id(has_background, use_full_label_map):
-        shift = 0 if has_background else 1
-        max_cat = 90 if use_full_label_map else 80
-        max_key = max_cat - shift
-        res_map = {i: i + shift for i in range(0, max_key+1)}
-        assert max(res_map.values()) == max_cat
-        return res_map
-
-    def _prepare_coco_structures(self):
-        annotation_conversion_parameters = self.dataset.config.get('annotation_conversion')
-        if not annotation_conversion_parameters:
-            raise ConfigError('annotation_conversion parameter is not pointed, '
-                              'but it is required for coco original metrics')
-        annotation_file = annotation_conversion_parameters.get('annotation_file')
-        if not annotation_file.is_file():
-            raise ConfigError("annotation file '{}' is not found".format(annotation_file))
-        has_background = annotation_conversion_parameters.get('has_background', False)
-        use_full_label_map = annotation_conversion_parameters.get('use_full_label_map', False)
+    def _prepare_coco_structures(self, annotations):
         if not self.dataset.metadata:
```

Comment:
from my view, the check that meta provided and label map inside should be moved to configure()
the metric calculated after full dataset processing, it will be very disappointing if you spend a lot of time to wait whole dataset inference and it failed that you do not provide meta throw several hours...
Configure called before modell running.

Reply:
Please review again